core/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -80,6 +81,8 @@
 
     private ActiveShardCount waitForActiveShards = ActiveShardCount.DEFAULT;
 
+    private transient CustomPrototypeRegistry registry = CustomPrototypeRegistry.EMPTY;
```

Comment:
> I feel like this should be a parameter to the source method.

Initially I didn't want change any method signatures, but it isn't a big deal after all as like you said just `null` can be provided. To make it even better we can add an overloaded version of the method that just accepts source.

> Is the reason you are considering a "named writeable to xcontent" so we don't have to have these registry objects?

Yes, here and also in [other](https://github.com/elastic/elasticsearch/pull/21243/files#diff-511bcea6f80008e3668e1361c6f9ced6R44) [places](https://github.com/elastic/elasticsearch/pull/21243/files#diff-1e645a531384a824ae99442770359b35R34) too.